src/main/java/wooteco/subway/auth/infrastructure/LoginInterceptor.java

```diff
@@ -0,0 +1,30 @@
+package wooteco.subway.auth.infrastructure;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import wooteco.subway.auth.application.AuthService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class LoginInterceptor implements HandlerInterceptor {
+    private AuthService authService;
+
+    public LoginInterceptor(AuthService authService) {
+        this.authService = authService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        if (isPreflight(request)) {
+            return true;
+        }
+        String credentials = AuthorizationExtractor.extract(request);
+        authService.validateToken(credentials);
+        return HandlerInterceptor.super.preHandle(request, response, handler);
```

Comment:
`HandlerInterceptor.super.preHandle(request, response, handler)`는 무조건 true를 반환하니 아래와 같이 사용해도 좋을 것 같아요ㅎㅎ
```suggestion
        return authService.validateToken(credentials);
```

Reply:
너무 IDE의 자동 완성을 무비판적으로 수용했네요😭. [수정](https://github.com/tributetothemoon/atdd-subway-fare/commit/7a4145206cdeda1ce7feece7b9bcef605b90a244)했습니다!!